src/RulesEngine/Validators/RuleValidator.cs

```diff
@@ -39,7 +39,9 @@ private void RegisterExpressionTypeRules()
         {
             When(c => c.Operator == null && c.RuleExpressionType == RuleExpressionType.LambdaExpression, () => {
                 RuleFor(c => c.Expression).NotEmpty().WithMessage(Constants.LAMBDA_EXPRESSION_EXPRESSION_NULL_ERRMSG);
-                RuleFor(c => c.Rules).Empty().WithMessage(Constants.LAMBDA_EXPRESSION_RULES_ERRMSG);
+                When(c => c.Operator == null, () => RuleFor(c => c.Rules).Empty().WithMessage(Constants.OPERATOR_RULES_ERRMSG)).Otherwise(() => {
+                    RuleFor(c => c.Rules).Empty().WithMessage(Constants.LAMBDA_EXPRESSION_RULES_ERRMSG);
+                });
```

Comment:
i think you can skip the Otherwise condition completely. Since the above when condition ensures Operator is always null.

Also can you add a small test case to validate this change?